design/Forest_Plot_design.Rmd

```diff
@@ -0,0 +1,340 @@
+---
+title: "Forest Plot design to present unadjusted and adjusted results"
+output:
+  html_document:
+    code_folding: hide
+    theme: spacelab 
+    toc: true
+    number_sections: true
+---
+
+  
+<style type="text/css">
+  
+body{ /* Normal  */
+      font-size: 14px;
+  }
+td {  /* Table  */
+    font-size: 14px;
+}
+h1.title {
+  font-size: 28px;
+  color: DarkRed;
+}
+h1 { /* Header 1 */
+    font-size: 22px;
+  color: DarkBlue;
+}
+h2 { /* Header 2 */
+    font-size: 18px;
+  color: DarkBlue;
+}
+h3 { /* Header 3 */
+    font-size: 18px;
+  color: DarkBlue;
+}
+
+h4 { /* Header 4 */
+    font-size: 18px;
+  color: DarkBlue;
+}
+
+code.r{ /* Code block */
+    font-size: 12px;
+}
+pre { /* Code block - determines code spacing between lines */
+    font-size: 14px;
+}
+</style>
+  
+  
+  
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  fig.width = 10,
+  fig.height = 6
+)
+``` 
+
+
+# Load libraries
+```{r}
+library(gridExtra)
+library(ggplot2)
+```
+
+# Background 
+
+Design forest plot to visualize the direction and magnitude of the unadjusted and adjusted results for each category of endpoints.
+We would have a separate forest plot including all the time to event endpoints, binary endpoints etc.
+
+In this example I have used the overall survival and more endpoints could be added (e.g. PFS etc.) in the same plot.
+An error bar with a diamond will be presented to indicate the point estimate with the corresponding 95% confidence interval for the OS unadjusted and adjusted results. Next to the forest plot and in the same line with the error bars we will present two columns that will show the coresponding HR [95% CI] and corresponding p-value. 
+
+
+# Steps
+
+- Load data 
+- Modify the report_table function in the matching_unanchored_tte.R file to get the results with a separate column for the HR and the 95% CI
+- In the maic_tte_unanchor function of the matching_unanchored_tte.R file, call the modified report_table so that the modified results are printed
+- Run the modified maic_tte_unanchor function to get the unadjusted and adjusted results 
```

Comment:
I don't think that this would have an impact to other functions.

About adding a test, would the following be fine?
```
test_that("Printing forest plot actually works",{
  p <- maic_forest_plot(tte_unanchor_res)
  expect_error(print(p), NA)

})
```

Reply:
Yes, we need a test for the new plot, but we should also add some tests the for functions like maic_tte_unanchor() that are modified. No need to add them here, we can do it in a separate issue.